### PR TITLE
offer MYSQL_COMPATIBILITY setting to remove unique from the model

### PR DESCRIPTION
**Pull request by tuky**
Fixes https://github.com/xtrinch/fcm-django/issues/266. Funny enough I just saw the commit with the swappable model and it seems possible to do this now:

```
class CustomFCMDevice(AbstractFCMDevice):
    registration_id = models.CharField(  # https://github.com/xtrinch/fcm-django/issues/266
        verbose_name="Registration token",
        unique=True,
        max_length=600,  # https://stackoverflow.com/a/64902685 better to be safe than sorry
    )

    class Meta(OriginalFCMDevice.Meta):
        pass
```

with setting `FCM_DJANGO_FCMDEVICE_MODEL = '...CustomFCMDevice'`.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -380,6 +380,19 @@ logins on the same device, you do not wish the old user to receive messages whil
 Via DRF, any creation of device with an already existing registration ID will be transformed into an update.
 If done manually, you are responsible for deleting the old device entry.
 
+MySQL compatibility
+-------------------
+MySQL has a limit for indices and therefore the `registration_id` field cannot be made unique in MySQL.
+We detect the database backend and remove the unique constraint for MySQL in the migration files. However,
+to ensure that the constraint is removed from the actual model you have to add the following to your settings:
```
Comment (xtrinch) [inline docs/index.rst:387]: Should make it clear somehow this is only an edge case and most people even if using mysql won't need this (and mention what the use case would be)

File: fcm_django/migrations/0011_fcmdevice_fcm_django_registration_id_user_id_idx.py [hunk 2]
```diff
@@ -33,12 +35,16 @@ class Migration(migrations.Migration):
...
+                model_name="fcmdevice",
+                index=models.Index(
+                    fields=["registration_id", "user"],
+                    name="fcm_django__registr_dacdb2_idx",
+                ),
             ),
-        ),
-    ]
+        ]
+        if not SETTINGS["MYSQL_COMPATIBILITY"]
```
Comment (xtrinch) [inline fcm_django/migrations/0011_fcmdevice_fcm_django_registration_id_user_id_idx.py:48]: Why explicitly check this? It should skip it if it's mysql anway?

Comment (tuky) [reply fcm_django/migrations/0011_fcmdevice_fcm_django_registration_id_user_id_idx.py:48]: To make `makemigrations` happy:

1st execution of `makemigrations` is with the check.
2nd execution is without it.

image

File: docs/index.rst (hunk 1 above)
Comment (tuky) [reply docs/index.rst]: improved it and explained my alternative solution that even lets mysql users enjoy an indexed `registration_id`

**Comment by tuky** [comment]
btw, i would be happy with only merging the readme part regarding a custom model and ditch the code & setting part. for us the custom model is enough as soon as it's released.

**Comment by xtrinch** [comment]
I wouldn't want everyone to have to migrate to custom model because of this and there seems to be a lot of people having issues with this.

**Comment by xtrinch** [comment]
Fixes #266, #258

**Review by xtrinch — APPROVED** [review]

**Pull request merged**